platform/posix/transport/src/transport_reconnect_posix.c

```diff
@@ -20,50 +20,52 @@
  */
 
 /**
- * @file reconnect_posix.c
+ * @file transport_reconnect_posix.c
  * @brief Implementation of the backoff logic when connection fails to the server fails.
  */
 
 /* Standard includes. */
 #include <unistd.h>
 #include <stdlib.h>
 #include <time.h>
-#include "reconnect.h"
+#include <math.h>
+
+#include "transport_reconnect.h"
 
 /*-----------------------------------------------------------*/
 
 bool Transport_ReconnectBackoffAndSleep( TransportReconnectParams_t * pReconnectParams )
 {
     bool status = false;
-    uint32_t jitter = 0;
 
     /* If MAX_RECONNECT_ATTEMPTS is set to 0, try forever */
-    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPS ) ||
-        ( 0 == MAX_RECONNECT_ATTEMPS ) )
+    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPTS ) ||
+        ( 0 == MAX_RECONNECT_ATTEMPTS ) )
     {
-        /*  Wait for timer to expire for the next reconnect */
-        ( void ) sleep( pReconnectParams->reconnectTimeoutSec );
+        /*  Wait for backoff time to expire for the next reconnect. */
+        ( void ) sleep( ( rand() % pReconnectParams->nextJitterMax ) );
```

Comment:
These kind of nested calls become difficult to debug when stepping through. I'd like to store the value of `rand() % pReconnectParams->nextJitterMax` in a variable before calling sleep.

Reply:
Updated